### PR TITLE
Visual Indicator for aimbot and triggerbot + gamesense tag (need some help)

### PR DESCRIPTION
**Pull request by newcommerdontblame**
gs tag is probably worst way to do but it should work theoretically xd((numbers are completely random so u can edit urself))

### CODE REVIEW

**Comment by newcommerdontblame** [comment]
https://streamable.com/xg7jxk

**Comment by shvshlik** [comment]
it doesnt even look like real one

**Comment by newcommerdontblame** [comment]
> it doesnt even look like real one

it looks pretty same to me 
like i said numbers are random u can make it faster or slower if u think its not same

**Comment by shvshlik** [comment]
its not about timings, the whole animation is wrong. also why tf would u use cases

**Comment by shvshlik** [comment]
"whole", it should start with ga

**Comment by newcommerdontblame** [comment]
it should be fix

**Comment by newcommerdontblame** [comment]
ok it should be fixed now

**Comment by Cyk-Fad** [comment]
old shadowware feature xd

**Comment by newcommerdontblame** [comment]
> old shadowware feature xd

Ahahha yes

File: Osiris/Hacks/Visuals.cpp [hunk 1]
```diff
@@ -406,3 +406,78 @@ void Visuals::skybox(FrameStage stage) noexcept
...
+            aimbotIndicator = aimbotIndicator + L"AIM";
+
+            std::wstring triggerbotIndicator;
+            triggerbotIndicator = triggerbotIndicator + L"TRIGGER";
+
+            if (config->visuals.selectedIndicators[0])
+            {
+                interfaces->surface->setTextFont(62); // aimbot indicator
+                interfaces->surface->setTextPosition(bottomLeft[0], bottomLeft[1] - (screenSizeMultiplier[1] * 75) - aimbotHeight);
+                if (config->aimbot.enabled)
```
Comment (newcommerdontblame) [inline Osiris/Hacks/Visuals.cpp:464]: need to fix here

File: Osiris/Hacks/Visuals.cpp [hunk 2]
```diff
@@ -406,3 +406,78 @@ void Visuals::skybox(FrameStage stage) noexcept
...
+                else
+                    interfaces->surface->setTextColor(255, 0, 0, 255);
+                interfaces->surface->printText(aimbotIndicator);
+            }
+
+            if (config->visuals.selectedIndicators[1])
+            {
+                interfaces->surface->setTextFont(62); // triggerbot indicator
+                interfaces->surface->setTextPosition(bottomLeft[0], bottomLeft[1] - (screenSizeMultiplier[1] * 75) - triggerbotHeight);
+                if (config->triggerbot.enabled)
```
Comment (newcommerdontblame) [inline Osiris/Hacks/Visuals.cpp:475]: need to fix here too

**Comment by newcommerdontblame** [comment]
I just wanna make this like in feature

if aimbot off = "AIM" indicator red
if aimbot on  
          if its only aimlock = shows FOV in indicator but green
          if its silent aim = shows FOV but its red

if trigger bot off = "TRIGGER'' indicator red 
if trigger bot on = shows triggerbot delay like "50ms" in green

**Comment by newcommerdontblame** [comment]
could you help please @danielkrupinski @Cyk-Fad

**Comment by SapphireCodenz** [comment]
image

**Comment by newcommerdontblame** [comment]
@SapphireCodenz it isnt finished yet

**Comment by techzin** [comment]
try this
if (&config->aimbot)

**Comment by aljawary** [comment]
> @newcommerdontblame
> 
> here try this
> 
> ```
> void Misc::gamesenseTag(bool tagChanged) noexcept
> {
>     if (config->misc.gamesenseTag) {
>         static int prevtime;
>         static int prevval;
>         if (int(memory->globalVars->currenttime * 2.4) != prevtime)
>             {
>             if (prevval != int(memory->globalVars->currenttime * 2.4) % 27)
>             {
>                 prevval = int(memory->globalVars->currenttime * 2.4) % 27;
>                 switch (int(memory->globalVars->currenttime * 2.4) % 27) {
>                 case 0: memory->setClanTag("                  ","                  "); break;
>                 case 1: memory->setClanTag("                 g","                 g"); break;
>                 case 2: memory->setClanTag("                ga","                ga"); break;
>                 case 3: memory->setClanTag("               gam","               gam"); break;
>                 case 4: memory->setClanTag("              game","              game"); break;
>                 case 5: memory->setClanTag("             games","             games"); break;
>                 case 6: memory->setClanTag("            gamese","            gamese"); break;
>                 case 7: memory->setClanTag("           gamesen","           gamesen"); break;
>                 case 8: memory->setClanTag("          gamesens","          gamesens"); break;
>                 case 9: memory->setClanTag("         gamesense","         gamesense"); break;
>                 case 10:memory->setClanTag("        gamesense ","        gamesense "); break;
>                 case 11:memory->setClanTag("       gamesense  ","       gamesense  "); break;
>                 case 12:memory->setClanTag("      gamesense   ","      gamesense   "); break;
>                 case 13:memory->setClanTag("     gamesense    ","     gamesense    "); break;
>                 case 14:memory->setClanTag("    gamesense     ","    gamesense     "); break;
>                 case 15:memory->setClanTag("   gamesense      ","   gamesense      "); break;
>                 case 16:memory->setClanTag("  gamesense       ","  gamesense       "); break;
>                 case 17:memory->setClanTag(" gamesense        "," gamesense        "); break;
>                 case 18:memory->setClanTag("gamesense         ","gamesense         "); break;
>                 case 19:memory->setClanTag("amesense          ","amesense          "); break;
>                 case 20:memory->setClanTag("mesense           ","mesense           "); break;
>                 case 22:memory->setClanTag("esense            ","esense            "); break;
>                 case 23:memory->setClanTag("sense             ","sense             "); break;
>                 case 24:memory->setClanTag("sens              ","sens              "); break;
>                 case 25:memory->setClanTag("sen               ","sen               "); break;
>                 case 26:memory->setClanTag("se                ","se                "); break;
>                 case 27:memory->setClanTag("s                 ","s                 "); break;
>                 default:;
>                 }
>             }
>         }
>         prevtime = int(memory->globalVars->currenttime);
>     }
> }
> ```

i think this is way better

**Comment by newcommerdontblame** [comment]
Hey @danielkrupinski every cheat has indicators like ths if you can add fov indicator for aimbot and shot ms for triggerbot too this would be cool 
(if u want u can remove gs tag)

**Pull request closed without merging**